src/SfxWeb/src/app/modules/event-store/event-store/event-store.component.ts

```diff
@@ -177,6 +181,17 @@ export class EventStoreComponent implements OnInit, OnDestroy {
       });
   }
 
+  processData(option: IOptionData){
+    if (option.addToList){
+      this.listEventStoreData.push(option.data);
+    }
+    else{
+      const index = this.listEventStoreData.indexOf(option.data);
```

Comment:
change this to the filter list operation, I think something like this. condense it to 1 line and use the build in list operation.
this.listEventStoreData = this.listEventStoreData.filter(item => item !== option.data)